src/Common/src/System/Security/Cryptography/RSACng.EncryptDecrypt.cs

```diff
@@ -225,34 +225,38 @@ private unsafe byte[] EncryptOrDecrypt(SafeNCryptKeyHandle key, ReadOnlySpan<byt
 
             byte[] output = new byte[estimatedSize];
             int numBytesNeeded;
-            ErrorCode errorCode = encrypt ?
-                Interop.NCrypt.NCryptEncrypt(key, input, input.Length, paddingInfo, output, output.Length, out numBytesNeeded, paddingMode) :
-                Interop.NCrypt.NCryptDecrypt(key, input, input.Length, paddingInfo, output, output.Length, out numBytesNeeded, paddingMode);
+            ErrorCode errorCode =
+                EncryptOrDecrypt(key, input, output, paddingMode, paddingInfo, encrypt, out numBytesNeeded);
 
             if (errorCode == ErrorCode.NTE_BUFFER_TOO_SMALL)
             {
+                CryptographicOperations.ZeroMemory(output);
```

Comment:
Can you remind me why this is necessary?  Windows will have filled this with the decrypted data as part of finding out that it was too small?

Reply:
> Windows will have filled this with the decrypted data as part of finding out that it was too small?

That's what we were seeing when they returned accidental success.  It might not be needed in "legitimate" NTE_BUFFER_TOO_SMALL cases, but since this is clearly an area that has had recent change I don't know we could even look to past behavior to predict future results.